website/common/script/ops/unlock.js

```diff
@@ -1,18 +1,69 @@
 import get from 'lodash/get';
-import each from 'lodash/each';
-import pick from 'lodash/pick';
 import setWith from 'lodash/setWith';
 import i18n from '../i18n';
-import splitWhitespace from '../libs/splitWhitespace';
-import {
-  NotAuthorized,
-  BadRequest,
-} from '../libs/errors';
+import { NotAuthorized, BadRequest } from '../libs/errors';
 
 import { removeItemByPath } from './pinnedGearUtils';
 import getItemInfo from '../libs/getItemInfo';
 import content from '../content/index';
 
+const incentiveBackgrounds = ['blue', 'green', 'red', 'purple', 'yellow'];
+
+function determineCost (isBackground, isFullSet) {
+  if (isBackground) {
+    return isFullSet ? 3.75 : 1.75;
+  }
+  return isFullSet ? 1.25 : 0.5;
+}
+
+function isGear (path) {
+  return path.includes('gear.');
+}
+
+function alreadyOwns (user, path) {
+  return get(user, isGear(path) ? path : `purchased.${path}`);
```

Comment:
I think this is correct for non gear items but for gear, you can actually lose an unlocked gear item in which case `user.items.gear.owned[key]` will be `false`  but you still own it (can buy it back with gold and not gems) so I would check for the key missing (user has never owned the gear):

```
return isGear(path) 
    ? get(user, path) !== undefined;
    : get(user, `purchased.${path}`) === true; // so that it's converted to a boolean
```